File: PrimaMateria/PrimaMateria/Classes/XTRGraphChoiceViewController.m
//
//  XTRGraphChoiceViewController.m
//  PrimaMateria
//
//  Created by Jerry Porter on 3/13/10.
//  Copyright 2016 xTrensa. All rights reserved.
//

#import "PrimaMateria.h"

@implementation XTRGraphChoiceViewController

#pragma mark - Action Methods

- (IBAction)chooseGraph: (id)sender {
    [[NSNotificationCenter defaultCenter] postNotificationName: NOTIFICATION_GRAPH_SELECTED object: @([sender tag])];
}

#pragma mark - View Management Methods

- (BOOL) shouldAutorotateToInterfaceOrientation: (UIInterfaceOrientation) interfaceOrientation {
    return UIInterfaceOrientationIsLandscape(interfaceOrientation);
}

#pragma mark - Memory Management Methods

@end

File: PrimaMateria/PrimaMateria/Classes/XTRGraphChoiceViewController.swift
//
//  XTRGraphChoiceViewController.swift
//  PrimaMateria
//
//  Created by Jerry Porter on 4/15/16.
//  Copyright © 2016 xTrensa. All rights reserved.
//

@objc class XTRGraphChoiceViewController : UIViewController {
    
    // MARK: - Initialization Methods
    
    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)!
    }
    
    // MARK: - Internal Methods
    
    // MARK: - Action Methods
    
    @IBAction func chooseGraph(sender: UIButton) {
        NSNotificationCenter.defaultCenter().postNotificationName(NOTIFICATION_GRAPH_SELECTED, object: sender.tag)
    }
    
    // MARK: - View Management Methods
        
    override func shouldAutorotate() -> Bool {
        return true
    }
    
    override func supportedInterfaceOrientations() -> UIInterfaceOrientationMask {
        return [UIInterfaceOrientationMask.LandscapeLeft, UIInterfaceOrientationMask.LandscapeRight]
    }
    
    // MARK: - Memory Management Methods
}
